src/platform/nm-platform-utils.c

```diff
@@ -404,7 +455,18 @@ nmp_utils_ethtool_set_link_settings (int ifindex,
 	edata.cmd = ETHTOOL_SSET;
 	if (autoneg) {
 		edata.autoneg = AUTONEG_ENABLE;
-		edata.advertising = edata.supported;
+		if (!speed && !duplex)
+			edata.advertising = edata.supported;
+		else {
+			unsigned long int mode;
+
+			mode = get_baset_mode (speed, duplex);
+
+			if (   !mode
+			    || !(edata.supported & mode))
+				return FALSE;
```

Comment:
This returns FALSE silently, without possibility to find out why it failed in the logfile. ethool-utils already log with level trace when something important happens. It should be done in this case as well.

Reply:
Right. Added.